src/Commands/Extension/AzCommand.cs

```diff
@@ -72,43 +64,28 @@ protected override AzArguments BindArguments(ParseResult parseResult)
 
     private static string? FindAzCliPath()
     {
+        string executableName = "az";
+
         // Return cached path if available and still exists
         if (!string.IsNullOrEmpty(_cachedAzPath) && File.Exists(_cachedAzPath))
         {
             return _cachedAzPath;
         }
 
-        var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
-        var searchPaths = new List<string>(AzureCliPaths);
-
-        // Add PATH environment directories
-        if (Environment.GetEnvironmentVariable("PATH")?.Split(Path.PathSeparator) is { } pathDirs)
-        {
-            searchPaths.AddRange(pathDirs);
-        }
+        var pathEnv = Environment.GetEnvironmentVariable("PATH");
+        if (string.IsNullOrEmpty(pathEnv))
+            return null;
 
-        foreach (var dir in searchPaths.Where(d => !string.IsNullOrEmpty(d)))
+        string[] paths = pathEnv.Split(Path.PathSeparator);
+        foreach (string path in paths)
         {
-            if (isWindows)
-            {
-                var cmdPath = Path.Combine(dir, "az.cmd");
-                if (File.Exists(cmdPath))
-                {
-                    _cachedAzPath = cmdPath;
-                    return cmdPath;
-                }
-            }
-            else
+            string fullPath = Path.Combine(path.Trim(), executableName);
+            if (File.Exists(fullPath))
```

Comment:
There is an edge case where the running process would not have access to the path and this would throw an `UnauthorizedAccessException`

Reply:
There are valid scenarios where we would run an az command.

One case is using the CLI credential; as you mentioned, if that fails, it should raise an UnauthorizedAccessException.

Another scenario is running the az CLI command as an extension.
In this case, raising an UnauthorizedAccessException would not be appropriate.

Given that, I believe it should be up to the caller to decide which exception to throw.